float32 ratio
bool rotation0
bool rotation1
bool rotation2
bool rotation3
uint32 transactionId